pkg/controllers/provisioning/suite_test.go

```diff
@@ -145,6 +145,18 @@ var _ = Describe("Provisioning", func() {
 				ExpectScheduled(ctx, env.Client, pod)
 			}
 		})
+		Context("Instance Type Selection", func() {
+			It("should not select invalid instance types", func() {
+				pod := ExpectProvisioned(ctx, env.Client, selectionController, provisioningController, provisioner, test.UnschedulablePod(
+					test.PodOptions{
+						ResourceRequirements: v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("3"), v1.ResourceMemory: resource.MustParse("3Gi")}},
+					},
+				))[0]
```

Comment:
I'm not sure what the output is if ExpectedProvisioned returns 0 pods, but I'm guessing it's a not very graceful crash. Would it be more informative to the user to add an explicit:

```
Expect(len(pods)).To(Equal(1))
node := ExpectScheduled(ctx, env.Client, pod[0])
```